project/StatusPlugin.scala

```diff
@@ -1,20 +1,29 @@
 import sbt._
 import Keys._
-import java.util.regex.Pattern
 
 object StatusPlugin extends AutoPlugin {
   override def requires = plugins.JvmPlugin
   override def trigger = allRequirements
 
   object autoImport {
-    lazy val publishStatus = SettingKey[String]("publish-status")
+    lazy val publishStatus = settingKey[String]("possible values are snapshots or releases.")
   }
 
   import autoImport._
 
   override def buildSettings: Seq[Setting[_]] = Seq(
-    isSnapshot <<= version(v => v.contains("-") && snapshotQualifier(v)),
-    publishStatus <<= isSnapshot { snap => if (snap) "snapshots" else "releases" },
+    isSnapshot := {
+      val SnapshotQualifier = """(.+)(-.*SNAPSHOT)(.*)""".r
+      val v = version.value
+      v match {
+        case SnapshotQualifier(_, _, _) => true
+        case _                          => false
+      }
```

Comment:
Yea. I aped the original implementation but likely `version.value contains "-SNAPSHOT"` would work just as well.